sunpy/net/helio/tests/test_helio.py

```diff
@@ -259,16 +260,39 @@ def test_link_test_on_urlerror(mock_link_test):
     link_test('') is None
 
 
+# Check if a server is active or not using the isOpen functionality
+def isOpen(ip, port, timeout):
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.settimeout(timeout)
+    try:
+        s.connect((ip, int(port)))
+        s.shutdown(socket.SHUT_RDWR)
+        return True
+    except socket.gaierror:
+        return False
+    finally:
+        s.close()
+
+
 @pytest.mark.remote_data
 @pytest.fixture(scope="session")
 def client():
     working_links = list(filter(link_test, webservice_parser()))
     taverna_link = taverna_parser(working_links[0])[0]
-    return HECClient(taverna_link)
+    # first check if the server is active
+    ip, port = taverna_link.split('/')[2].split(':')
+    server_status = isOpen(ip, port, 3)
+    if(server_status):
+        return HECClient(taverna_link)
+    else:
+        return None
```

Comment:
If this return was just `pytest.xfail` would that work?

Reply:
I guess pytest.xfail has to be called for test functions so that those funcitons depending on its data do not fail. because client is not failing in any way its just not catching any data which could be normal... while its a error for those relying on this data

Reply:
If you did `pytest.xfail("server broke")` would that not work instead of returning None?